snuba/clusters/cluster.py

```diff
@@ -137,37 +141,46 @@ def __init__(
         super().__init__(storage_sets)
         if not single_node:
             assert cluster_name
-        self.__host = host
-        self.__port = port
+
+        self.__query_node = ClickhouseNode(host, port)
         self.__http_port = http_port
         self.__single_node = single_node
         self.__cluster_name = cluster_name
         self.__distributed_cluster_name = distributed_cluster_name
         self.__reader: Optional[Reader[SqlQuery]] = None
         self.__connection_cache: MutableMapping[
-            ClickhouseClientSettings, ClickhousePool
+            Tuple[ClickhouseNode, ClickhouseClientSettings], ClickhousePool
         ] = {}
 
     def __str__(self) -> str:
-        return f"{self.__host}:{self.__port}"
+        return str(self.__query_node)
 
     def get_connection(
-        self, client_settings: ClickhouseClientSettings,
+        self,
+        client_settings: ClickhouseClientSettings,
+        node: Optional[ClickhouseNode] = None,
     ) -> ClickhousePool:
         """
         Get a Clickhouse connection using the client settings provided. Reuse any
-        connection to the cluster with the same settings otherwise establish a new
+        connection to the same node with the same settings otherwise establish a new
         connection.
+
+        If no node is passed, we get a connection to the query node.
         """
-        if client_settings not in self.__connection_cache:
-            settings, timeout = client_settings.value
-            self.__connection_cache[client_settings] = ClickhousePool(
-                self.__host,
-                self.__port,
+        if not node:
```

Comment:
```
if node is not none
```
since you are checking for the difference between the object being present or none and you. do not want to look into whether the object is truthy or not.
PEP8 recommendation (I learnt recently).

Reply:
Nice